crates/items/src/item_stack.rs

```diff
@@ -0,0 +1,237 @@
+#![forbid(unsafe_code, warnings)]
+
+use crate::{Enchantment, Item};
+use core::fmt;
+use serde::de::{Error, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Represents an item stack.
+///
+/// An item stack includes an item type, an amount and a bunch of properties (enchantments, etc.)
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct ItemStack {
+    /// The item type of this `ItemStack`.
+    #[serde(rename = "id")]
+    pub item: Item,
+
+    /// The number of items in the `ItemStack`.
+    pub count: u32,
+
+    /// The `ItemStack` metadata, containing data such as damage,
+    /// repair cost, enchantments...
+    #[serde(rename = "tag")]
+    pub meta: ItemStackMeta,
```

Comment:
Maybe this should be an `Option`

Reply:
Yes, my bad. The NBT tag `tag` is not always present. Will fix it now.